src/device/router_device_channels_worker.erl

```diff
@@ -544,7 +546,7 @@ maybe_report_downlink_dropped(DeviceID, Desc, Channel) ->
     end.
 
 -spec downlink_decode(binary() | map()) ->
-    {ok, {boolean(), integer(), binary()} | clear_queue} | {error, any()}.
+    {ok, {boolean(), integer(), atom(), binary()} | clear_queue} | {error, any()}.
```

Comment:
This could be your region type here instead of `atom()`
```suggestion
    {ok, {boolean(), integer(), downlink_region(), binary()} | clear_queue} | {error, any()}.
```

Reply:
Addressed.